### ISSUE TITLE
Media Gallery Shift+Click Selection in Reverse Order isn't working as expected

### ISSUE THREAD

**Issue by anefzaoui** [Automattic/wp-calypso#62618]
### Quick summary

Expected is the keyword here, I am not sure if the current behavior is the intended behavior, but I'm seeing that people come to expect a different one.

When choosing images for a Tiled Galley Block, in the Media Gallery, When you shift+select a batch of media items from the end to the start, the order should remain End To Start, but that's not the case.
What happens is that if you want to Select 5 images with Shift+Click and start from the end, the expected order result should be: [ 5 ] [ 4 ] [ 3 ] [ 2 ] [ 1 ]
the current order result is: [ 2 ] [ 3 ] [ 4 ] [ 5 ] [ 1 ]

Here's a GIF of the bug:

![IncorrectOrderBugInMediaGallery - Copy](https://user-images.githubusercontent.com/1832078/162412496-bfe6e539-c825-4b82-bf3e-5da4c2741cff.gif)

### Steps to reproduce

Create a new post
Add Tiled Gallery Block
Select images and choose Media Library as source
Have 5 media items present
Shift+Select from old to new

### What you expected to happen

Correct order from older to newer,
or [ 5 ] [ 4 ] [ 3 ] [ 2 ] [ 1 ]

### What actually happened

Oldest is first, then order resets to newer to older.
Or: [ 2 ] [ 3 ] [ 4 ] [ 5 ] [ 1 ]

### Simple, Atomic or both?

Simple, Atomic

### Reproducibility

Consistent

### Severity

All

### Available workarounds?

Yes, difficult to implement

### Workaround details

The workaround is manual selection.
This sounds fine until you have to select a couple of hundreds of media items.

**Comment by anefzaoui** [comment]
`4934186-zn-woo` to follow up on once a PR with a fix is merged into trunk.

**Comment by renata-franco** [comment]
The report came from this ticket: 4934186-zd-woothemes
Updated some of the things in the bug report! But everything looks great 😄

**Comment by Robertght** [comment]
Thank you for reporting this!

The Tiled Gallery block is a Jetpack block: https://jetpack.com/support/jetpack-blocks/

I can replicate this on a self-hosted environment


https://user-images.githubusercontent.com/6737345/163985402-65d6d569-9091-464d-a656-449e6b3458c7.mp4

I wasn't able to find a report about this, but I'm wondering if this is worth moving it over to the Jetpack repo for now. cc @jeherve
